File: app/src/main/java/mb/pokequiz/PokeApplication.java
package mb.pokequiz;

import android.app.Application;

import mb.pokequiz.dagger.ApplicationComponent;
import mb.pokequiz.dagger.ApplicationModule;
import mb.pokequiz.dagger.DaggerApplicationComponent;

/**
 * Created by mbpeele on 2/1/16.
 */
public class PokeApplication extends Application {

    private ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();
        component = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();
    }

    public ApplicationComponent getApplicationComponent() {
        return component;
    }
}

File: app/src/main/java/mb/pokequiz/PokeApplication.kt
package mb.pokequiz

import android.app.Application
import io.realm.Realm
import mb.pokequiz.dagger.app.*

/**
 * Created by mbpeele on 2/1/16.
 */
class PokeApplication : Application() {

    companion object {
        @JvmStatic lateinit var component: AppComponent
    }

    override fun onCreate() {
        super.onCreate()

        Realm.init(this)

        component = DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .persistenceModule(PersistenceModule())
                .webModule(WebModule())
                .build()
    }


}
